test/integration/enumerate_databases.test.ts

```diff
@@ -0,0 +1,79 @@
+import { expect } from 'chai';
+
+import { AddUserOptions, MongoClient, MongoServerError } from '../../src';
+
+describe('listDatabases', function () {
+  // TODO(NODE3860): Create driver test variants that run with AUTH enabled
+  beforeEach(function () {
+    if (process.env.AUTH !== 'auth') {
+      this.skip();
+    }
+  });
+
+  describe('authorizedDatabases flag', function () {
+    const username = 'a';
+    const password = 'b';
+    const mockAuthorizedDb = 'mockAuthorizedDb';
+
+    let client: MongoClient;
+    let authorizedClient: MongoClient;
+
+    const authorizedUserOptions: AddUserOptions = {
+      roles: [{ role: 'read', db: mockAuthorizedDb }]
+    };
+
+    beforeEach(async function () {
+      // pass credentials from cluster_setup's mlaunch defaults
+      // TODO(NODE3860): pass credentials instead based on environment variable
+      client = this.configuration.newClient({ auth: { username: 'user', password: 'password' } });
+      await client.connect();
+
+      await client
+        .db('mockAuthorizedDb')
+        .createCollection('a')
+        .catch(() => null);
+
+      await client.db('admin').addUser(username, password, authorizedUserOptions);
+
+      authorizedClient = this.configuration.newClient({
+        auth: { username: username, password: password }
+      });
+      await authorizedClient.connect();
+    });
+
+    afterEach(async function () {
+      await client.db('admin').removeUser(username);
+      await client.close();
+      await authorizedClient.close();
+    });
+
+    it('should list authorized databases for authorizedDatabases set to true', async function () {
+      const adminListDbs = await client.db().admin().listDatabases();
+      const authorizedListDbs = await authorizedClient
+        .db()
+        .admin()
+        .listDatabases({ authorizedDatabases: true });
+      const adminDbs = adminListDbs.databases;
+      const authorizedDbs = authorizedListDbs.databases;
+
+      expect(adminDbs).to.have.length.greaterThan(1);
+      expect(authorizedDbs).to.have.lengthOf(1);
+
+      expect(adminDbs.filter(db => db.name === mockAuthorizedDb).length).equals(1);
+      expect(adminDbs.filter(db => db.name !== mockAuthorizedDb).length).greaterThan(0);
+      expect(authorizedDbs.filter(db => db.name === mockAuthorizedDb).length).equals(1);
+    });
+
+    it('should error for authorizedDatabases set to false', async function () {
```

Comment:
is this test supposed to be for the case where there are no authorized databases? is so, that should be part of the test case description, because setting the flag itself to false should not cause an error; we should have tests here that show a successful use of the authorizedDatabases flag set to false as well as a case for the flag being omitted entirely

Reply:
yeah this was for the case of existing authorized databases, with the flag set to false -- sorry updated the test descriptions to be more specific